### DISCUSSION TITLE
fallback route

### DISCUSSION THREAD

**Discussion by soxft** [zeromicro/go-zero#2905] [Q&A] [answered]
Is there any way to custom api fallback route?

---

```go
server := rest.MustNewServer(c.RestConf, rest.WithNotFoundHandler(handler.NotAllowHandler()))
```

**Comment by kevwan** [comment, marked as answer]
SetNotFoundHandler

**Reply by soxft** [reply]
<img width="619" alt="image" src="https://user-images.githubusercontent.com/42080379/220849335-25f353e7-15a6-4623-83fb-51ed11050516.png">

但是这样访问不存在的路由时 依旧返回 404 not found, 但是 notAllowedHandler 有效
